tests/python/gaia-ui-tests/gaiatest/tests/functional/lockscreen/test_lockscreen_time_check.py

```diff
@@ -0,0 +1,84 @@
+from marionette.wait import Wait
+
+from gaiatest.apps.settings.app import Settings
+from gaiatest import GaiaTestCase
+from gaiatest.apps.lockscreen.app import LockScreen
+
+
+class TestLockScreen(GaiaTestCase):
+
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+        self.connect_to_local_area_network()
+
+    def test_lockscreen_time_check(self):
+        """
+        https: // bugzilla.mozilla.org / show_bug.cgi?id = 1118054
+        """
+
+        self.settings = Settings(self.marionette)
+        self.settings.launch()
+        datetime = self.settings.open_date_and_time_settings()
+
+        # Auto time update is by default set to true, turn it off to make region change
+        datetime.toggle_automatic_time_update()
+        self.assertFalse(datetime.is_autotime_enabled, 'Autotime still enabled')
+
+        # record time and change the region
+        old_time = datetime.get_current_time_text
+        datetime.set_region('Atlantic Ocean')
+
+        # get the displayed time after the region change
+        new_time = datetime.get_current_time_text
+        self.assertNotEqual(new_time, old_time)
+
+        # lock screen and check time on the lockscreen
+        self.marionette.switch_to_frame()
+        self.device.lock()
+        lock_screen = LockScreen(self.marionette)
+        self.assertLessEqual(self.get_time_difference(new_time, lock_screen.time), 1)
+
+        # configure to set the time automatically (this will revert the timezone change), then lock screen
+        lock_screen.switch_to_frame()
+        lock_screen.unlock()
+        self.apps.switch_to_displayed_app()
+
+        # Enable the auto time update, so the regions change back and date/time is reverted back
+        datetime.toggle_automatic_time_update()
+        self.assertTrue(datetime.is_autotime_enabled, 'Autotime still disabled')
+        self.marionette.switch_to_frame()
+        self.device.lock()
+
+        # wait until device is off and turn back on to check that the time is changed
+        Wait(self.marionette, timeout=20).until(
+            lambda m: not self.device.is_screen_enabled)
+        self.device.turn_screen_on()
+
+        # Check it reverted to the correct time, and compare it with the previously shown time
+        # Allow 5 minutes difference max
+        self.assertLessEqual(self.get_time_difference(old_time, lock_screen.time), 5)
+
+    def get_time_difference(self, old_time, new_lockscreen_time):
```

Comment:
The following comment got lost in the PR:

``` Python
# in Lockscreen 
    @property
    def time(self):
        return time.strptime(self.marionette.find_element(*self._time_locator).text, '%M:%S %p')
        # AM/PM is hidden in a span, we would be able to get it

# Same thing in settings/date time

# in test_lockscreen_time_check.py 
new_time = datetime.time
delta = new_time - lockscreen.time
self.assertLessEqual(delta.total_seconds(), 5)
```

Do you think this would be doable?